Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryRecorderFactory.cs

```diff
@@ -63,14 +64,27 @@ internal static class OpenTelemetryRecorderFactory
                         config: requestOptions?.CosmosThresholdOptions ?? clientContext.ClientOptions?.CosmosClientTelemetryOptions.CosmosThresholdOptions);
                 }
 #if !INTERNAL
-                else if (Activity.Current is null)
+                // The scope here checks for listeners at Network Level.
+                // If there are listeners at network level and no parent activity created at operation level
+                // then create a network scope that creates a parent activity.
+                else
                 {
                     DiagnosticScope requestScope = LazyNetworkScopeFactory.Value.CreateScope(name: operationName);
+                    openTelemetryRecorder = requestScope.IsEnabled ? OpenTelemetryCoreRecorder.CreateNetworkLevelParentActivity(networkScope: requestScope) : openTelemetryRecorder;
+                }
 
-                    openTelemetryRecorder = requestScope.IsEnabled ? OpenTelemetryCoreRecorder.CreateNetworkLevelParentActivity(networkScope: requestScope) : OpenTelemetryCoreRecorder.CreateParentActivity(operationName);
+                // If there are no listeners at operation level and network level and no parent activity
+                // then create a dummy parent activity.
+                if (Activity.Current is null)
+                {
+                    openTelemetryRecorder = OpenTelemetryCoreRecorder.CreateParentActivity(operationName);
                 }
 #endif
-                trace.AddDatum("DistributedTraceId", Activity.Current?.TraceId);
+                // Safety check as diagnostic logs should not break the code.
+                if (Activity.Current?.TraceId != null)
+                {
+                    trace.AddDatum("DistributedTraceId", Activity.Current?.TraceId);
```

Comment:
At this point, nothing is `null`, all 3 items were checked, right?

```suggestion
                    trace.AddDatum("DistributedTraceId", Activity.Current.TraceId);
```

Reply:
removed